searx/engines/apple_app_store.py

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""
+    Apple App Store
+"""
+
+from json import loads
+from urllib.parse import urlencode
+from dateutil.parser import parse
+
+about = {
+    "website": 'https://www.apple.com/app-store/',
+    "wikidata_id": 'Q368215',
+    "official_api_documentation": (
+        'https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/'
+        'iTuneSearchAPI/UnderstandingSearchResults.html#//apple_ref/doc/uid/TP40017632-CH8-SW1'
+    ),
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+categories = ['files', 'apps']
+safesearch = True
+
+search_url = 'https://itunes.apple.com/search?{query}'
+
+
+def request(query, params):
+    explicit = "Yes"
+
+    if params['safesearch'] > 0:
+        explicit = "No"
+
+    params['url'] = search_url.format(query=urlencode({'term': query, 'media': 'software', 'explicit': explicit}))
+
+    return params
+
+
+def response(resp):
+    results = []
+
+    json_result = loads(resp.text)
+
+    for result in json_result['results']:
+        results.append(
+            {
+                'url': result['trackViewUrl'],
+                'title': result['trackName'],
+                'content': result['description'],
+                'img_src': result['artworkUrl100'],
+                'thumbnail': result['screenshotUrls'][0],
```

Comment:
The other engines in the `app` category doesn't provide a screenshot : 
![image](https://user-images.githubusercontent.com/1594191/186968533-6afdf054-fbec-42f1-bb73-ae93eb178bda.png)

This make the "apple_app_store" engine huge compare to others.

So, can you remove the thumbnail?

Reply:
I thought it'd be nice if we could get a quick peek of the app before we actually open the URL, but ok

Reply:
done

Reply:
> I thought it'd be nice if we could get a quick peek of the app before actually opening the URL

I agree; the thing is, I don't know how to display these screenshots.
I thought about something similar to the image: when the users click on a result, the screenshot(s) are displayed on the right side, so the users can still browse the result quickly without too much information in the result list.